### ISSUE TITLE
[Bug]: Quest theme file not working correctly

### ISSUE THREAD

**Issue by aaronhowser1** [FTBTeam/FTB-Mods-Issues#482]
### Mod

FTB Quests

### Mod version

ftb-quests-forge-1802.3.8-build.101

### Forge / Fabric version

Forge 1.18.2 40.1.60

### What issue are you having?

```txt
[*]
background:          ftblibrary:textures/gui/background_squares.png; color=#DCFFFFFF; tile_size=64
extra_quest_shapes:  diamond, pentagon, hexagon, octagon, heart, gear

text_color:          #FFFFFF
hover_text_color:    #FFFFA0
disabled_text_color: #999999

widget_border:     #1B1D1E
widget_background: #44000000
symbol_in:         #9BC600
symbol_out:        #408300

button:                hollow_rectangle:{{widget_border}}
panel:                 {{container_slot}}
disabled_button:       hollow_rectangle:{{widget_border}}
hover_button:          {{button}} + {{widget_background}}; padding=1
context_menu:          hollow_rectangle:{{widget_border}}; round_edges + {{background}}; color=#B4B4B4; padding=1
scroll_bar_background: {{widget_background}}
scroll_bar:            {{button}} + {{widget_background}}; padding=1
container_slot:        {{button}}; padding=-1
text_box:              part:minecraft:textures/gui/container/enchanting_table.png; pos=0,185,108,19; corner=6

icon:                 none
check_icon:           builtin
add_icon:             builtin
alert_icon:           ftbquests:textures/gui/alert.png
support_icon:         ftblibrary:textures/icons/support.png
wiki_icon:            ftbquests:textures/gui/info.png
wiki_url:             -
pin_icon_on:          ftbquests:textures/gui/pin.png; color=#F8BD1C
pin_icon_off:         ftbquests:textures/gui/pin.png; color=#DBDBDB
settings_icon:        ftbquests:textures/gui/settings.png
close_icon:           ftbquests:textures/gui/close.png
emergency_items_icon: ftbquests:textures/gui/emergency_items.png; color=#F3D9FF
guide_icon:           ftbquests:textures/gui/guide.png; color=#EF9E1A
modpack_icon:         ftbquests:textures/item/book.png
reward_table_icon:    ftblibrary:icons/money_bag
shop_icon:            ftbquests:textures/gui/shop.png; color=#EF9E1A
collect_rewards_icon: ftbquests:textures/gui/collect_rewards.png; color=#EF9E1A
delete_icon:          ftblibrary:icons/remove
reload_icon:          ftblibrary:icons/refresh
download_icon:        ftblibrary:icons/down
edit_icon:            ftblibrary:icons/settings
move_up_icon:         ftblibrary:icons/up
move_down_icon:       ftblibrary:icons/down

full_screen_quest:                  0
tasks_text_color:                   #5555FF
rewards_text_color:                 #FFAA00
quest_view_background:              {{context_menu}}
quest_view_border:                  {{widget_border}}
quest_view_title:                   #AAAAAA
quest_completed_color:              #C856FF56
quest_started_color:                #C800FFFF
dependency_line_texture:            ftbquests:textures/gui/dependency.png
dependency_line_completed_color:    #64DC64
dependency_line_requires_color:     #00C8C8
dependency_line_required_for_color: #C8C800
dependency_line_selected_speed:     1.0
dependency_line_unselected_speed:   0.0
dependency_line_thickness:          0.17
quest_spacing:                      1.0
pinned_quest_size:                  0.75

checkmark_task_active:   ftblibrary:icons/accept
checkmark_task_inactive: ftblibrary:icons/accept_gray

[561D507841C34CB0]
quest_completed_color:              #C8FF00E5
quest_started_color:                #C87700FF
```

![image](https://user-images.githubusercontent.com/15698722/193406064-583eb1f6-db22-4740-93e9-adf53a01cc2f.png)

The border should be this color
![image](https://user-images.githubusercontent.com/15698722/193406179-7e84c872-4020-4554-980f-8aeb608c2012.png)
but it isn't

### Steps to reproduce

N/A

**Comment by desht** [comment]
Long->int quest ID truncation issue when reading the theme file

**Comment by aaronhowser1** [comment]
Still nonfunctional in the dev version we got for sb3

**Comment by desht** [comment]
What's the exact file name of the mod you're running?  It's definitely working here:
![2022-10-17_08 52 42](https://user-images.githubusercontent.com/736831/196120109-e90ebf75-fbb1-4546-b9fe-5a86e02ea9e0.png)
```
$ cat resourcepacks/ftb_theme/assets/ftbquests/ftb_quests_theme.txt 
[51FD605E11A11AB4]
quest_completed_color:              #C8FF00E5
quest_started_color:                #C87700FF
```

**Comment by aaronhowser1** [comment]
![image](https://user-images.githubusercontent.com/15698722/196124681-6c304a87-f316-4560-bf48-9fe16612ec8b.png)

```txt
[*]
background:          ftblibrary:textures/gui/background_squares.png; color=#DCFFFFFF; tile_size=64
extra_quest_shapes:  diamond, pentagon, hexagon, octagon, heart, gear

text_color:          #FFFFFF
hover_text_color:    #FFFFA0
disabled_text_color: #999999

widget_border:     #1B1D1E
widget_background: #44000000
symbol_in:         #9BC600
symbol_out:        #408300

button:                hollow_rectangle:{{widget_border}}
panel:                 {{container_slot}}
disabled_button:       hollow_rectangle:{{widget_border}}
hover_button:          {{button}} + {{widget_background}}; padding=1
context_menu:          hollow_rectangle:{{widget_border}}; round_edges + {{background}}; color=#B4B4B4; padding=1
scroll_bar_background: {{widget_background}}
scroll_bar:            {{button}} + {{widget_background}}; padding=1
container_slot:        {{button}}; padding=-1
text_box:              part:minecraft:textures/gui/container/enchanting_table.png; pos=0,185,108,19; corner=6

icon:                 none
check_icon:           builtin
add_icon:             builtin
alert_icon:           ftbquests:textures/gui/alert.png
support_icon:         ftblibrary:textures/icons/support.png
wiki_icon:            ftbquests:textures/gui/info.png
wiki_url:             -
pin_icon_on:          ftbquests:textures/gui/pin.png; color=#F8BD1C
pin_icon_off:         ftbquests:textures/gui/pin.png; color=#DBDBDB
settings_icon:        ftbquests:textures/gui/settings.png
close_icon:           ftbquests:textures/gui/close.png
emergency_items_icon: ftbquests:textures/gui/emergency_items.png; color=#F3D9FF
guide_icon:           ftbquests:textures/gui/guide.png; color=#EF9E1A
modpack_icon:         ftbquests:textures/item/book.png
reward_table_icon:    ftblibrary:icons/money_bag
shop_icon:            ftbquests:textures/gui/shop.png; color=#EF9E1A
collect_rewards_icon: ftbquests:textures/gui/collect_rewards.png; color=#EF9E1A
delete_icon:          ftblibrary:icons/remove
reload_icon:          ftblibrary:icons/refresh
download_icon:        ftblibrary:icons/down
edit_icon:            ftblibrary:icons/settings
move_up_icon:         ftblibrary:icons/up
move_down_icon:       ftblibrary:icons/down

full_screen_quest:                  0
tasks_text_color:                   #5555FF
rewards_text_color:                 #FFAA00
quest_view_background:              {{context_menu}}
quest_view_border:                  {{widget_border}}
quest_view_title:                   #AAAAAA
quest_completed_color:              #C856FF56
quest_started_color:                #C800FFFF
dependency_line_texture:            ftbquests:textures/gui/dependency.png
dependency_line_completed_color:    #64DC64
dependency_line_requires_color:     #00C8C8
dependency_line_required_for_color: #C8C800
dependency_line_selected_speed:     1.0
dependency_line_unselected_speed:   0.0
dependency_line_thickness:          0.17
quest_spacing:                      1.0
pinned_quest_size:                  0.75

checkmark_task_active:   ftblibrary:icons/accept
checkmark_task_inactive: ftblibrary:icons/accept_gray

[561D507841C34CB0]
quest_completed_color:              #C8FF00E5
quest_started_color:                #C87700FF
```

Tested with [561D507841C34CB0] at the top and the bottom

**Comment by desht** [comment]
And what's the filename of the FTB Quests jar?

**Comment by aaronhowser1** [comment]
ftb-quests-forge-1802.3.9-build.119

**Comment by desht** [comment]
That's weird then. Either kubejs isn't loading that resource file correctly, or you have some kind of mismatch between the quest ID in the resource file and the loaded quest.  As I said, it definitely works in my dev instance, using a theme file loaded from a vanilla-style (not kubejs) resource pack. Can you test with a resource pack loaded as a vanilla resource pack?

**Comment by aaronhowser1** [comment]
![image](https://user-images.githubusercontent.com/15698722/196809232-c6dd85a3-2cf4-4f45-95b4-45879fe98cdb.png)

```
[*]
background:          ftblibrary:textures/gui/background_squares.png; color=#DCFFFFFF; tile_size=64
extra_quest_shapes:  diamond, pentagon, hexagon, octagon, heart, gear

text_color:          #FFFFFF
hover_text_color:    #FFFFA0
disabled_text_color: #999999

widget_border:     #1B1D1E
widget_background: #44000000
symbol_in:         #9BC600
symbol_out:        #408300

button:                hollow_rectangle:{{widget_border}}
panel:                 {{container_slot}}
disabled_button:       hollow_rectangle:{{widget_border}}
hover_button:          {{button}} + {{widget_background}}; padding=1
context_menu:          hollow_rectangle:{{widget_border}}; round_edges + {{background}}; color=#B4B4B4; padding=1
scroll_bar_background: {{widget_background}}
scroll_bar:            {{button}} + {{widget_background}}; padding=1
container_slot:        {{button}}; padding=-1
text_box:              part:minecraft:textures/gui/container/enchanting_table.png; pos=0,185,108,19; corner=6

icon:                 none
check_icon:           builtin
add_icon:             builtin
alert_icon:           ftbquests:textures/gui/alert.png
support_icon:         ftblibrary:textures/icons/support.png
wiki_icon:            ftbquests:textures/gui/info.png
wiki_url:             -
pin_icon_on:          ftbquests:textures/gui/pin.png; color=#F8BD1C
pin_icon_off:         ftbquests:textures/gui/pin.png; color=#DBDBDB
settings_icon:        ftbquests:textures/gui/settings.png
close_icon:           ftbquests:textures/gui/close.png
emergency_items_icon: ftbquests:textures/gui/emergency_items.png; color=#F3D9FF
guide_icon:           ftbquests:textures/gui/guide.png; color=#EF9E1A
modpack_icon:         ftbquests:textures/item/book.png
reward_table_icon:    ftblibrary:icons/money_bag
shop_icon:            ftbquests:textures/gui/shop.png; color=#EF9E1A
collect_rewards_icon: ftbquests:textures/gui/collect_rewards.png; color=#EF9E1A
delete_icon:          ftblibrary:icons/remove
reload_icon:          ftblibrary:icons/refresh
download_icon:        ftblibrary:icons/down
edit_icon:            ftblibrary:icons/settings
move_up_icon:         ftblibrary:icons/up
move_down_icon:       ftblibrary:icons/down

full_screen_quest:                  0
tasks_text_color:                   #5555FF
rewards_text_color:                 #FFAA00
quest_view_background:              {{context_menu}}
quest_view_border:                  {{widget_border}}
quest_view_title:                   #AAAAAA
quest_completed_color:              #C856FF56
quest_started_color:                #C800FFFF
dependency_line_texture:            ftbquests:textures/gui/dependency.png
dependency_line_completed_color:    #64DC64
dependency_line_requires_color:     #00C8C8
dependency_line_required_for_color: #C8C800
dependency_line_selected_speed:     1.0
dependency_line_unselected_speed:   0.0
dependency_line_thickness:          0.17
quest_spacing:                      1.0
pinned_quest_size:                  0.75

checkmark_task_active:   ftblibrary:icons/accept
checkmark_task_inactive: ftblibrary:icons/accept_gray

[2FAF298DB97FAFCF]
quest_completed_color:              #C8FF00E5
quest_started_color:                #C87700FF
```

![image](https://user-images.githubusercontent.com/15698722/196811332-ddece22d-d8b0-43f6-ad8a-ff2b1b1e9283.png)

**Comment by aaronhowser1** [comment]
> $ cat resourcepacks/ftb_theme/assets/ftbquests/ftb_quests_theme.txt 
> [51FD605E11A11AB4]
> quest_completed_color:              #C8FF00E5
> quest_started_color:                #C87700FF

Is this the entire file? Should I have the [*] bit at all?

**Comment by desht** [comment]
You don't strictly need it, unless you're overriding the default settings (the `[*]` section applies to all quest objects).  Anything in the custom theme file is merged into the base theme that the mod ships as default, so sections with a specific quest id (or tag) can override the properties of that quest or quests.

But it shouldn't cause any problems to have it there.  Really not sure why it's working for you, I've been through the code again and can't see why it wouldn't work after the fix I made (and it does work here).

**Comment by aaronhowser1** [comment]
***I hadn't changed the *unlocked but unstarted color****, *just the started and the completed colors*. God damn it.
![image](https://user-images.githubusercontent.com/15698722/197132374-ad333127-73c7-487e-887c-f3aee3cd1c00.png)

How would I change the unlocked but unstarted color?

**Comment by aaronhowser1** [comment]
I'm assuming started is 1-99% completion, completed is obv 100%

**Comment by desht** [comment]
Looking at the code, yeah - started means the progress has been updated to anything >0.  Right now it looks like the "available but unstarted" color is hardcoded to white...

**Comment by aaronhowser1** [comment]
I don't suppose this has been changed?

**Comment by desht** [comment]
It has been changed - sorry, thought I'd told you!  There are two now entries in `ftb_quests_theme.txt` in the current release which you can override as needed:

```
quest_not_started_color:            #96FFFFFF
quest_locked_color:                 #FF999999
```

**Comment by desht** [comment]
Implemented for 1.18 & 1.19
